src/components/study/perimeter/StudyFlowViewer.tsx

```diff
@@ -0,0 +1,52 @@
+'use client'
+
+import PdfViewer from '@/components/document/PDFViewer'
+import { Document } from '@prisma/client'
+import { useTranslations } from 'next-intl'
+import Image from 'next/image'
+import { useEffect, useMemo, useState } from 'react'
+import styles from './StudyFlow.module.css'
+
+interface Props {
+  selectedFlow: Document
+  documentUrl?: string
+}
+
+const StudyFlowViewer = ({ documentUrl, selectedFlow }: Props) => {
+  const [width, setWidth] = useState(0)
+  const [height, setHeight] = useState(0)
+  const t = useTranslations('study.perimeter')
+
+  const setDimensions = async (src: string) => {
+    const image = new window.Image()
+    image.onload = () => {
+      setWidth(image.width)
+      setHeight(image.height)
+    }
+    image.src = src
+  }
```

Comment:
Compliqué ce code. Il sert a quoi ?

Reply:
J'utilise next/image plutôt qu'une simple balise <img> (warning de next  :`Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` or a custom image loader to automatically optimize images.`

Le composant <Image> nécessite width et height que je ne connais pas d'avance. Il faut donc les calculer, et c'est à ça que cette fonction sert. J'ai déplacé le code dans un composant ImageViewer dédié pour plus de clareté

Reply:
Tu peux donner la width maximal / height de ton composant :)
et la tu pers l'interet de l'optimisation, vu que tu loades l'image, tu recuperes sa width, puis tu loades l'Image